app/views/comments/_form.html.erb

```diff
@@ -6,6 +6,7 @@
   <%= f.text_area :body %> <br>
   <%= f.hidden_field :commentable_id, value: comment.commentable.id %>
   <%= f.hidden_field :commentable_type, value: comment.commentable.class %>
+  <%= f.hidden_field :ancestry unless comment.ancestry.blank? %>

```

Comment:
I don't like it either, but can't find better solution. Changing that field will generate wrong link in notification message, it's bad but not dangerous.